lxd/device/usb.go

```diff
@@ -170,18 +170,18 @@ func (d *usb) startVM() (*deviceConfig.RunConfig, error) {
 	runConf := deviceConfig.RunConfig{}
 	runConf.PostHooks = []func() error{d.Register}
 
+	var nMatched int
 	for _, usb := range usbs {
-		if !usbIsOurDevice(d.config, &usb) {
-			continue
+		if usbIsOurDevice(d.config, &usb) {
+			runConf.USBDevice = append(runConf.USBDevice, deviceConfig.USBDeviceItem{
+				DerivedDeviceName: fmt.Sprintf("%s-%d", d.name, nMatched),
+				HostDevicePath:    fmt.Sprintf("/dev/bus/usb/%03d/%03d", usb.BusNum, usb.DevNum),
+			})
+			nMatched++
 		}
-
-		runConf.USBDevice = append(runConf.USBDevice, []deviceConfig.RunConfigItem{
-			{Key: "devName", Value: d.name},
-			{Key: "hostDevice", Value: fmt.Sprintf("/dev/bus/usb/%03d/%03d", usb.BusNum, usb.DevNum)},
-		}...)
 	}
 
-	if d.isRequired() && len(runConf.USBDevice) <= 0 {
```

Comment:
I dont think we need to change this part, if no devices match then runConf.USBDevice will be empty.

Reply:
Hmm, I was under the impression that it can be something that starts non-empty as it was matched by earlier devices? If you have two usb-device pass through, first one matched some devices, second did not but is required, this condition would be wrong, right?

Reply:
No I don't think so, there is a new `RunConfig` create for each logical LXD USB device starting up on line 170, then you are only adding matching actual devices to the `runConf.USBDevice` field, so if none have matched it will be empty.

Reply:
Ah, thanks for clarifying that. Adjusted.

Shall I squash history and rewrite the commit message to make things more consistent?

Reply:
Yes please